go/pkg/apis/paddlepaddle/v1alpha1/types.go

```diff
@@ -0,0 +1,89 @@
+package v1alpha1
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+)
+
+const (
+	// CRDKind original TPR name is TraingingJob, but I think
+	// it is too common and easy to conflict
+	CRDKind       = "PdJob"
+	CRDKindPlural = "pdjobs"
+	CRDGroup      = "paddlepaddle.org"
+	CRDVersion    = "v1alpha1"
+)
+
+// CRDName returns name of crd
+func CRDName() string {
+	return fmt.Sprintf("%s.%s", CRDKindPlural, CRDGroup)
+}
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PdJob is a specification for a PdJob resource
+type PdJob struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PdJobSpec   `json:"spec"`
+	Status PdJobStatus `json:"status"`
+}
+
+// PdJobSpec is the spec for a PdJob resource
+type PdJobSpec struct {
```

Comment:
This is similar to https://github.com/PaddlePaddle/cloud/blob/develop/go/api/types.go can those codes be reused or just move them?

Reply:
@typhoonzero we can also reused the codes and structs.